GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/ModifiedPathsTests.cs

```diff
@@ -172,15 +171,22 @@ public static object[] Runners
             {
                 get
                 {
-                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    List<object[]> hardLinkRunners = new List<object[]>();
+                    foreach (object[] runner in FileSystemRunner.Runners.ToList())
                     {
-                        return new[]
+                        FileSystemRunner fileSystem = runner.ToList().First() as FileSystemRunner;
+                        if (fileSystem.SupportsHardlinkCreation)
                         {
-                            new object[] { new CmdRunner() },
-                            new object[] { new BashRunner() },
-                        };
+                            hardLinkRunners.Add(new object[] { fileSystem });
+                        }
                     }
 
+                    if (hardLinkRunners.Count > 0)
+                    {
+                        return hardLinkRunners.ToArray();
+                    }
+
+                    // Always return at least one runner that supports creating hard links

```

Comment:
Isn't it a bug in our code if we get this far? I would just throw.

Reply:
@sanoursa this is not a bug because when we run without `--full-suite` the only runner will be `SystemIORunner` which does not support hard link creation.

I can either update the comment to include those details, or I can mark the test as full suite only, let me know if you have a preference.

Reply:
Got it, yea a comment would be helpful

Reply:
Actually, I just noticed that we had this same issue for `RenameDirectory` and addressed it by having `SystemIORunner.RenameDirectory` PInvoke to the native API.  I'll see if I can do the same for hard link creation and then all of this "does runner support hard link" code can be removed.